google-cloud-lib/src/main/java/com/streamsets/pipeline/stage/cloudstorage/destination/GoogleCloudStorageTarget.java

```
@@ -0,0 +1,128 @@
+    private Calendar calendar;
+    private CredentialsProvider credentialsProvider;
+
+    public GoogleCloudStorageTarget(GCSTargetConfig gcsTargetConfig) {
+        this.gcsTargetConfig = gcsTargetConfig;
+    }
+
+    /** {@inheritDoc} */
+    @Override
+    protected List<ConfigIssue> init() {
+        // Validate configuration values and open any required resources.
+        List<ConfigIssue> issues = gcsTargetConfig.init(getContext(), super.init());
+        gcsTargetConfig.credentials.getCredentialsProvider(getContext(), issues).ifPresent(p -> credentialsProvider = p);
+
+        storage = StorageOptions.getDefaultInstance().getService();
+
+        elVars = getContext().createELVars();
+        partitionEval = getContext().createELEval(PARTITION_TEMPLATE);
+
+        calendar = Calendar.getInstance(TimeZone.getTimeZone(gcsTargetConfig.timeZoneID));
+
+        try {
+            storage = StorageOptions.newBuilder().setCredentials(credentialsProvider.getCredentials()).build().getService();
+        } catch (IOException e) {
+            LOG.error(Errors.GCS_01.getMessage(), e.toString(), e);
+        }
+
+        return issues;
+    }
+
+    /** {@inheritDoc} */
+    @Override
+    public void destroy() {
+        // Clean up any open resources.
+        super.destroy();
+    }
+
+    /** {@inheritDoc} */
+    @Override
+    public void write(Batch batch) throws StageException {
+        ByteArrayOutputStream bOut = new ByteArrayOutputStream();
+
+        Iterator<Record> batchIterator = batch.getRecords();
+        while (batchIterator.hasNext()) {
+            Record record = batchIterator.next();
+
+            DataGeneratorFactory generatorFactory = gcsTargetConfig.dataGeneratorFormatConfig.getDataGeneratorFactory();
+            try (DataGenerator gen = generatorFactory.getGenerator(bOut)) {
+                gen.write(record);
+            } catch (IOException e) {
+                throw new StageException(Errors.GCS_00, e);
```

This will basically kill the pipeline, its probably better to throw an error record in here, getContext().ToError(record, GCS.00, e);